src/js/express.js

```diff
@@ -64,98 +87,147 @@ gpii.express.registerComponentLineage = function (childComponent, expressCompone
+ * @param path {String} - The path segment where we can fine the child components we need to wire in to ourselves.
+ *
+ * Wire a child component to its immediate descendants using the `use` method common to both `express` and all
+ * middleware.
+ *
+ */
 gpii.express.connectDirectDescendants = function (that, component, path) {
-    var descendants =  that.childrenByParent[path];
-
-    // This component has descendants, wire them in first
-    if (descendants !== undefined) {
-        // Wire together one level of routers and middleware, then recurse
-        for (var a = 0; a < descendants.length; a++) {
-            var childNickname  = descendants[a];
-            var childComponent = component[childNickname];
-            if (fluid.hasGrade(childComponent.options, "gpii.express.router")) {
-                // We have to wire our children in with our path to preserve relative pathing, so that their router will begin with our path.
-                component.router.use(component.options.path, childComponent.router);
-
-                // Recurse from here on down.
-                that.connectDirectDescendants(childComponent, path + "." + childNickname);
-            }
-            else if (fluid.hasGrade(childComponent.options, "gpii.express.middleware")) {
-                if (component.router) {
-                    // We have to wire our children in with our path to preserve relative pathing, so that their router will begin with our path.
-                    component.router.use(component.options.path, childComponent.checkMethod);
-                }
-                else {
-                    fluid.fail("A component must expose a router in order to work with child middleware components.");
-                }
-            }
+    var orderedNicknames = gpii.express.getOrderedNicknames(that.childrenByParent[path], component);
+
+    // This component has descendants, wire them in first.
+    fluid.each(orderedNicknames, function (childNickname) {
+        var childComponent = component[childNickname];
+
+        if (fluid.hasGrade(childComponent.options, "gpii.express.middleware")) {
+            gpii.express.wireMiddlewareToContainer(component.router, childComponent);
         }
-    }
 
-    // After our child components are in place, wire the router to itself.
-    //
-    // This must be done here because we want to give children the chance to own part of the path before we take the rest over.
-    //
-    // The path and method have to be used here so that parameters will be parsed correctly.
-    if (fluid.hasGrade(component.options, "gpii.express.router")) {
-        component.router[component.options.method](component.options.path, component.route);
+        if (fluid.hasGrade(childComponent.options, "gpii.express.router")) {
+            // Recurse from here on down.
+            that.connectDirectDescendants(childComponent, path + "." + childNickname);
```

Comment:
Don't need to necessarily fix this now, but should leave a TODO that this will fail in the near future when the framework supports periods in component names. This whole network of functions would be much safer if they operate with a "segs" array of path segments in place of "path" String